Tests/Rewriters/ConstraintsRewriterTests.cs

```diff
@@ -51,84 +52,138 @@ public void SetUp ()
         //language=C#
-        @"_mock.Do (Arg<ITestInterface>.Matches (_ => _ != null && object.Equals (_.A, 42)));")]
+        @"_mock.DoSomething (Arg<ITestInterface>.Matches (_ => _ != null && (object.Equals (_.A, 42))));")]
     [TestCase (
         //language=C#
-        @"_mock.Do (Arg<int[]>.Matches (Rhino.Mocks.Constraints.List.IsIn (2)));",
+        @"_mock.DoSomething (Arg<int[]>.Matches (Rhino.Mocks.Constraints.List.IsIn (2)));",
         //language=C#
-        @"_mock.Do (Arg<int[]>.Matches (_ => _.Contains (2)));")]
+        @"_mock.DoSomething (Arg<int[]>.Matches (_ => _.Contains (2)));")]
     [TestCase (
         //language=C#
-        @"_mock.Do (Arg<int[]>.Matches (Rhino.Mocks.Constraints.List.ContainsAll (new[] {1, 2})));",
+        @"_mock.DoSomething (Arg<int[]>.Matches (Rhino.Mocks.Constraints.List.ContainsAll (new[] {1, 2})));",
         //language=C#
-        @"_mock.Do (Arg<int[]>.Matches (_ => new[] {1, 2}.All (_.Contains)));")]
+        @"_mock.DoSomething (Arg<int[]>.Matches (_ => new[] {1, 2}.All (_.Contains)));")]
+    [TestCase (
+        //language=C#
+        @"_mock.DoSomething (Arg<ITestInterface>.Matches (Rhino.Mocks.Constraints.Property.Value (""A"", 42) & Rhino.Mocks.Constraints.Property.Value (""B"", 1337)));",
+        //language=C#
+        @"_mock.DoSomething (Arg<ITestInterface>.Matches (_ => _ != null && (object.Equals (_.A, 42) && object.Equals (_.B, 1337))));")]
+    [TestCase (
+        //language=C#
+        @"_mock.DoSomething (Arg<ITestInterface>.Matches (Rhino.Mocks.Constraints.Is.NotNull() & Rhino.Mocks.Constraints.Property.Value (""B"", 1337)));",
+        //language=C#
+        @"_mock.DoSomething (Arg<ITestInterface>.Matches (_ => _ != null && object.Equals (_.B, 1337)));")]
+    [TestCase (
+        //language=C#
+        @"
+_mock.DoSomething (
+  Arg<ITestInterface>.Matches (
+    Rhino.Mocks.Constraints.Property.Value (""A"", 42) 
+    & Rhino.Mocks.Constraints.Property.Value (""B"", 1337)));",
+        //language=C#
+        @"
+_mock.DoSomething (
+  Arg<ITestInterface>.Matches (
+    _ => _ != null 
+         && (object.Equals (_.A, 42) 
+         && object.Equals (_.B, 1337))));")]
+    [TestCase (
+        //language=C#
+        @"
+_mock.DoSomething (
+  Arg<ITestInterface>.Matches (
+    Rhino.Mocks.Constraints.Property.Value (""A"", 42) 
+    | Rhino.Mocks.Constraints.Property.Value (""B"", 1337)));",
+        //language=C#
+        @"
+_mock.DoSomething (
+  Arg<ITestInterface>.Matches (
+    _ => _ != null 
+         && (object.Equals (_.A, 42) 
+             || object.Equals (_.B, 1337))));")]
+    [TestCase (
+        //language=C#
+        @"
+_mock.DoSomething (
+  Arg<ITestInterface>.Matches (
+    Rhino.Mocks.Constraints.Is.NotNull() 
+    | Rhino.Mocks.Constraints.Property.Value (""A"", 42) 
+    | Rhino.Mocks.Constraints.Property.Value (""B"", 1337)));",
+        //language=C#
+        @"
+_mock.DoSomething (
+  Arg<ITestInterface>.Matches (
+    _ => _ != null 
```

Comment:
if `_ != null` is evaluated in the following nested expressions, there is no need to check it again.

Reply:
u sure?
if `_` is `null`, `_.A` would throw an exception if there is no additional null check at the beginning of the expression

Reply:
Whoops formulated it wrong, the inner `_ != null` is not necessary.